### ISSUE TITLE
issue on BinaryFocalLoss

### ISSUE THREAD

**Issue by quancore** [Hsuxu/Loss_ToolBox-PyTorch#12]
I have two questions:

1) Why you are applying sigmoid at the beginning of fow$rward call?
2) Is label smoothing is correct? Should we apply to target rather than output?

**Comment by shahzad-ali** [comment]
This code has several critical deficiencies and you have highlighted some of those.

> 1. Why you are applying sigmoid at the beginning of forward call?

There is no harm in assuming that the model @Hsuxu used gave logits as output (i.e. no sigmoid/softmax function was used) so the first thing one will do before calculating any loss is to use an activation function. 

Let's hope @Hsuxu will answer the rest of the question soon!
